src/v/cluster/rm_stm.cc

```diff
@@ -1632,15 +1633,26 @@ void rm_stm::maybe_rearm_autoabort_timer(time_point_type deadline) {
 }
 
 ss::future<tx::errc> rm_stm::abort_all_txes() {
+    static constexpr uint max_concurrency = 5u;
```

Comment:
I'd like to get @bashtanov's thoughts on this. This was originally added for migration purposes, but I vaguely recall a discussion in the PR about the potential unsafe iteration (no synchronization when this method is called). I reviewed that PR but its not the same PR that added this method. Alexey, do you remember anything about it? I'm not able to find that discussion.

Reply:
Found the discussion 

https://github.com/redpanda-data/redpanda/pull/26380#discussion_r2138740678